src/main/java/by/itech/lab/supplier/controller/UserController.java

```diff
@@ -1,29 +1,73 @@
 package by.itech.lab.supplier.controller;
 
-import by.itech.lab.supplier.auth.UserImpl;
-import by.itech.lab.supplier.dto.CustomerDto;
+import by.itech.lab.supplier.constant.ApiConstants;
+import by.itech.lab.supplier.dto.UserDto;
+import by.itech.lab.supplier.service.UserService;
 import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.web.PageableDefault;
 import org.springframework.http.HttpStatus;
-import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.web.bind.annotation.DeleteMapping;
 import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.ResponseStatus;
 import org.springframework.web.bind.annotation.RestController;
 
-import java.util.List;
-
-import static by.itech.lab.supplier.constant.ApiConstants.URL_COMPANIES;
-import static by.itech.lab.supplier.constant.ApiConstants.URL_USER;
+import javax.validation.Valid;
 
 @RestController
+@Slf4j
 @AllArgsConstructor
-@RequestMapping(URL_USER)
+@RequestMapping(ApiConstants.URL_USER)
 public class UserController {
 
+    @Autowired
+    private final UserService userService;
+
+
+    @GetMapping(ApiConstants.URL_ID_PARAMETER)
+    public UserDto getUser(@PathVariable Long id) {
+        log.debug("request to get User : {}", id);
+        if (userService.getUserWithAuthoritiesById(id).isPresent()) {
```

Comment:
hmm...
1. what about orElseThrow
2. do you know that you can return optional from the controller? 😃